autoload/sneak.vim

```diff
@@ -155,13 +155,22 @@ func! sneak#to(op, input, inputlen, count, register, repeatmotion, reverse, incl
     endif
   endfor
 
+  " This determines sneak#is_sneaking() is true
+  " It needs to be on even if there's no match,
+  " so that clever-S can still find a possible match in reverse direction
+  call s:attach_autocmds()
+
   if 0 == max(matchpos)
     if nudge
       call sneak#util#nudge(a:reverse) "undo nudge for t
     endif
 
     let km = empty(&keymap) ? '' : ' ('.&keymap.' keymap)'
 
+    " TODO: highlight clears one step too late after no match if it was initiated while on the last match
+    " HACK solution: eat the outer nested autocmd...
+    doautocmd CursorMoved
```

Comment:
@justinmk how reasonable is this as a solution? Seems like it could introduce side effects in the worst case...

Demo of the highlight edge case this solves:

`sTasj`

https://github.com/user-attachments/assets/70e5c380-1acf-458f-a1cb-a3ba82c6b848